File: lessons/lesson30/UITableViewDynamicCells/UITableViewDynamicCells/ViewController.m
//
//  ViewController.m
//  UITableViewDynamicCells
//
//  Created by Руслан on 04.11.16.
//  Copyright © 2016 Ruslan Timchenko. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - UITableViewDataSource

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    
    NSLog(@"numberOfSectionsInTableView");
    
    return [[UIFont familyNames] count];
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    
    NSLog(@"numberOfRowsInSection %ld", section);
    
    NSArray *familyNames = [UIFont familyNames];
    
    NSString *currentName = [familyNames objectAtIndex:section];
    
    return [[UIFont fontNamesForFamilyName:currentName] count];
}

- (nullable NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section {
    
    return [[UIFont familyNames] objectAtIndex:section];
    
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    
    NSLog(@"cellForRowAtIndexPath {%ld, %ld}", indexPath.section, indexPath.row);
    
    static NSString *identifier = @"cell";
    
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:identifier];
    
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:identifier];
        NSLog(@"Cell is created");
    } else {
        NSLog(@"Cell is reused");
    }
    
    NSArray *familyNames = [UIFont familyNames];
    
    NSString *currentName = [familyNames objectAtIndex:indexPath.section];
    
    NSString *fontName = [[UIFont fontNamesForFamilyName:currentName] objectAtIndex:indexPath.row];
    
    UIFont *font = [UIFont fontWithName:fontName size:16];
    
    cell.textLabel.font = font;
    cell.textLabel.text = fontName;
    
    return cell;
}

@end

File: Homework/Swift/HW26/HW26/ViewController.swift
//
//  ViewController.swift
//  HW26
//
//  Created by Руслан on 29.10.16.
//  Copyright © 2016 Ruslan Timchenko. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    @IBOutlet weak var draggingView: UIImageView!
    
    @IBOutlet weak var rotationSwitch: UISwitch!
    @IBOutlet weak var scaleSwitch: UISwitch!
    @IBOutlet weak var translationSwitch: UISwitch!
    
    @IBOutlet weak var segmentedControl: UISegmentedControl!
    @IBOutlet weak var speedAnimationLabel: UILabel!
    @IBOutlet weak var speedAnimationSlider: UISlider!
    

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

